### PR TITLE
Added Dev Containers

### PR DESCRIPTION
**Pull request by AkashS0510**

### CODE REVIEW

File: .devcontainer/DockerFile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Base image: Python with additional tools
+FROM python:3.9-slim
```
Comment (refeed) [inline .devcontainer/DockerFile:2]: Could you please switch this to 3.11?

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "Tirith Dev Container",
+  "dockerFile": "Dockerfile",
+  "context": "..",
+  "features": {
+    "ghcr.io/devcontainers/features/python:1": {
+      "version": "3.9"
```
Comment (refeed) [inline .devcontainer/devcontainer.json:7]: Also switch to 3.11

File: .devcontainer/DockerFile (hunk 1 above)
Comment (refeed) [reply .devcontainer/DockerFile:2]: Because we want to make sure that it is working with the latest stable Python version

Comment (AkashS0510) [reply .devcontainer/DockerFile]: @refeed Have changed them.

**Pull request closed without merging**